support cross-compilation by not using the cfg macro in build.rs

noonien opened the pull request:
Fixes #14

thomwiggers commented on implementations.yaml:
```diff
@@ -6,7 +6,7 @@ traits_version: 0.3.2
   kyber:
     version: 0.6.5
-    avx2_guard: 'cfg(all(not(disable_avx2), not(target_os = "windows"), not(target_os = "macos"), target_arch = "x86_64"))'
+    avx2_guard: '!is_windows && !is_macos && target_arch == "x86_64"'
```
This is also not good enough: if you have an old CPU this will lead to illegal instructions.

thomwiggers replied:
Actually I should first have coffee before deciding such things — the above isn't true.

thomwiggers reviewed (changes requested):
On a second look, this actually seems to work.

The only issue remaining seems to change ``DEFAULT_AVX2_GUARD`` in ``generate_implementations.py``.

noonien commented:
Opps, missed that.

It's fixed now. Also did a `cargo check` and `cargo test ` in the workspace to make sure everything compiles, and runs.

There is a SIGSEGV in `pqcrypto-classicmceliece`, however, the error exists in the old code as well.

thomwiggers reviewed (changes requested):
Sorry, I forgot to also ask that you do not remove the `disable_avx2` opt-in feature

thomwiggers commented:
> There is a SIGSEGV in pqcrypto-classicmceliece, however, the error exists in the old code as well.

Yeah, I think there's some issue in the upstream code but I had a hard time locking it down.

noonien commented:
I'm not exactly sure what the disable_avx opt-in feature is. Will it be passed using a cfg_attr macro? Because it's not defined in Config.tom. If so, are you sure that is the right place to use it like this? I'm not sure those config attrs will be available in build.rs.

thomwiggers commented:
Some good questions :sweat_smile:  I think I used it via `-C disable_avx2` for testing.

noonien commented:
Was that set in RUSTFLAGS?

thomwiggers commented:
I think so, yeah

noonien commented:
[RUSTFLAGS also doesn't work when cross-compiling](https://github.com/rust-lang/cargo/issues/4423). I'll add a feature to Cargo.toml.

thomwiggers commented:
It's still helpful to have, it allows me to test the non-AVX2 codepaths on an AVX2 machine.

noonien commented:
Fixed.

I added a feature in Cargo.toml for packages that support avx2. To run without avx2 support use:
```
cargo build --no-default-features
```

noonien commented:
Thanks, cheers!

:handshake:

thomwiggers commented:
Thanks for the contribution

noonien commented:
Any chance to get this change in a release?

thomwiggers commented:
I've run the script.

I also should update everything with PQClean changes, but I'm not sure if I'll have time for that anytime soon...
